content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md

```diff
@@ -347,57 +684,116 @@ kubectl apply -f my-crontab.yaml
-Kubernetes API server.
+{{< note >}}
+Defaulting is available as alpha since 1.15. The
+`CustomResourceDefaulting` feature must be enabled, which is the case automatically for many clusters for beta features. Please refer to the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) documentation for more information.
 
-[kubectl](/docs/reference/kubectl/overview) consumes the published schema to perform client-side validation
-(`kubectl create` and `kubectl apply`), schema explanation (`kubectl explain`) on custom resources.
-The published schema can be consumed for other purposes. The feature is Alpha in 1.14 and disabled by default.
-You can enable the feature using the `CustomResourcePublishOpenAPI` feature gate on the
-[kube-apiserver](/docs/admin/kube-apiserver):
+Defaulting also requires a structural schema and pruning. 
+{{< /note >}}
 
+Defaulting allows to specify default values in the [OpenAPI v3 validation schema](#validation):
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: crontabs.stable.example.com
+spec:
+  group: stable.example.com
+  versions:
+    - name: v1
+      served: true
+      storage: true
+  version: v1
+  scope: Namespaced
+  names:
+    plural: crontabs
+    singular: crontab
+    kind: CronTab
+    shortNames:
+    - ct
+  preserveUnknownFields: false
+  validation:
+   # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          properties:
+            cronSpec:
+              type: string
+              pattern: '^(\d+|\*)(/\d+)?(\s+(\d+|\*)(/\d+)?){4}$'
+              default: "5 0 * * *"
+            image:
+              type: string
+            replicas:
+              type: integer
+              minimum: 1
+              maximum: 10
+              default: 1
 ```
---feature-gates=CustomResourcePublishOpenAPI=true
+
+With this both `cronSpec` and `replicas` are defaulted:
+
+```yaml
+apiVersion: "stable.example.com/v1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+spec:
+  image: my-awesome-cron-image
 ```
 
-Custom resource validation schema will be converted to OpenAPI v2 schema, and
+leads to
+
+```yaml
+apiVersion: "stable.example.com/v1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+spec:
+  cronSpec: "5 0 * * *"
+  image: my-awesome-cron-image
+  replaces: 1
+```
+
+Note that defaulting happens on the object
+ 
+* in the request to the API server using the request version defaults
+* when reading from etcd using the storage version defaults
+* after mutating admission plugins with non-empty patches using the admission webhook object version defaults.
+
+Hence, adding a default to the schema will automatically result in the defaults being applied from getting or listing objects, e.g. via `kubectl get`. Note though that defaults are not automatically written to the persisted objects.
```

Comment:
might be clearer as "Note that defaults applied when reading data from etcd are not automatically written back to etcd; an update request via the API is required to persist those defaults back into etcd."

Reply:
fixed